components/Common/Pagination/useGetPageElements.tsx

```diff
@@ -0,0 +1,146 @@
+'use client';
+import Link from 'next/link';
+import { useIntl } from 'react-intl';
+
+import styles from './index.module.css';
+
+import type { PaginationProps } from '.';
+
+type ParsedPage = PaginationProps['pages'][number] & {
+  pageNumber: number;
+};
+
+function parsePages(pages: PaginationProps['pages']): ParsedPage[] {
+  return pages.map(({ url }, index) => ({
+    url,
+    pageNumber: index + 1,
+  }));
+}
+
+const ellipsis = (
+  <span aria-hidden="true" className={styles.ellipsis}>
+    ...
+  </span>
+);
+
+function createPageElements({
+  parsedPages,
+  currentPage,
+  intl,
+}: Pick<PaginationProps, 'currentPage'> & {
+  parsedPages: ParsedPage[];
+  intl: ReturnType<typeof useIntl>;
+}) {
+  return parsedPages.map(({ url, pageNumber }) => {
+    return (
+      <li
+        key={pageNumber}
+        aria-setsize={parsedPages.length}
+        aria-posinset={pageNumber}
+      >
+        <Link
+          prefetch={false}
+          href={url}
+          aria-label={intl.formatMessage(
+            {
+              id: 'components.common.pagination.pageLabel',
+            },
+            {
+              pageNumber,
+            }
+          )}
+          className={styles.listItem}
+          {...(pageNumber === currentPage && { 'aria-current': 'page' })}
+        >
+          <span>{pageNumber}</span>
+        </Link>
+      </li>
+    );
+  });
+}
+
+export const useGetPageElements = (
```

Comment:
I worry this whole function has little to none inline documentation of what each thing does.

Reply:
Thank you. As mentioned in this [other comment](https://github.com/nodejs/nodejs.org/pull/5971#discussion_r1349698931), I will try to add comments that explain the different logical steps of the code, as I thought the code was pretty self-explicative but I now see that that's not the case.

The logic of this component is inherently complex, though, so the code will be a bit complex to understand but comments can definitely help.

You can take a look at React MUI [Pagination component's code](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/usePagination/usePagination.js) from the material-ui repository, which I took as inspiration. In my opinion, that code is a bit more difficult to follow.

Reply:
> You can take a look at React MUI Pagination component's code from the material-ui repository, which I took as inspiration. In my opinion, that code is a bit more difficult to follow.

If that was used as inspiration, please add an inline comment mentioning that 🙇

Reply:
Good catch!

I will add a comment referencing the file in an upcoming commit. Thank you!